vp/node/views.py

```diff
@@ -227,6 +237,23 @@ def execute_node(request, node_id):
     except NodeException as e:
         return JsonResponse({e.action: e.reason}, status=500)
 
+@swagger_auto_schema(method='get',
+                     operation_summary='Gets the data frame at the executed node.',
+                     operation_description='Retrieves the state of data at that point in the graph.',
+                     responses={
+                         200: 'Data successfully retrieved'
+                     })
+@api_view(['GET'])
+def retrieve_data(request, node_id):
+    # TODO: currently hard coding the name of the file as workflow + node id;
+    #       will need to change the word workflow for the workflow name. Will also need to add some exceptions.
+    #
+    file_name = 'workflow-' + str(node_id)
```

Comment:
In the future maybe we should make the data file an attribute on the node to avoid re-deriving the filename multiple places

Reply:
I like this idea. Probably could define a setter that concatenates `workflow_name + node_id` so it would update whenever the workflow name changes.

Reply:
Correct, in the future the workflow object will have a name instance variable and we can use this to set the name of the file through a common static function that can be re-use for reading and writing from/to the file.